daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -879,25 +885,23 @@ private[lf] object SBuiltin {
 
   /** $checkPrecondition
     *    :: arg (template argument)
+    *    -> Unit
     *    -> Bool (false if ensure failed)
     *    -> Unit
     */
-  final case class SBCheckPrecond(templateId: TypeConName) extends SBuiltinPure(2) {
+  final case class SBCheckPrecond(templateId: TypeConName) extends SBuiltinPure(3) {
     override private[speedy] def executePure(args: util.ArrayList[SValue]): SUnit.type = {
-      if (
-        !(getSList(args, 1).iterator.forall(_ match {
-          case SBool(b) => b
-          case otherwise => crash(s"type mismatch SBCheckPrecond: expected SBool, got $otherwise")
-        }))
-      )
+      val _ = getSUnit(args, 1)
```

Comment:
(another nit picking) 
```suggestion
      discard(getSUnit(args, 1))
```